components/script/dom/element.rs

```diff
@@ -2974,3 +2991,53 @@ pub fn cors_setting_for_element(element: &Element) -> Option<CorsSettings> {
         }
     })
 }
+
+pub fn lang_check(css_lang: &Box<str>, element_lang: String) -> bool {
```

Comment:
This doesn't need to be here; it can live in `components/script/dom/bindings/str.rs`.

Also, the parameters can just be `tag: &str, range: &str`. It's also a good idea to link to https://tools.ietf.org/html/rfc4647#section-3.3.2 above the function to indicate that this is following a specification.

Finally, the function name can just be `extended_filtering`.